// @cleartax/truss/dist/components/FilterSidebar/components/FilterOptionsCustomRenderers.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE_classnames__ from "classnames";
import * as __WEBPACK_EXTERNAL_MODULE__types_js_8472748c__ from "../types.js";
import * as __WEBPACK_EXTERNAL_MODULE__store_js_add3cf67__ from "../store.js";
import * as __WEBPACK_EXTERNAL_MODULE__renderers_SearchRenderer_js_43846080__ from "../renderers/SearchRenderer.js";
import * as __WEBPACK_EXTERNAL_MODULE__renderers_CheckboxFilterRenderer_js_9adad8cc__ from "../renderers/CheckboxFilterRenderer.js";
import * as __WEBPACK_EXTERNAL_MODULE__renderers_RadioFilterRenderer_js_6bf7a591__ from "../renderers/RadioFilterRenderer.js";
import * as __WEBPACK_EXTERNAL_MODULE__renderers_DateRangeFilterRenderer_js_61fc7460__ from "../renderers/DateRangeFilterRenderer.js";
import * as __WEBPACK_EXTERNAL_MODULE__renderers_MonthRangeFilterRenderer_js_ecc61406__ from "../renderers/MonthRangeFilterRenderer.js";
import * as __WEBPACK_EXTERNAL_MODULE__renderers_SliderFilterRenderer_js_3470f8b6__ from "../renderers/SliderFilterRenderer.js";
import * as __WEBPACK_EXTERNAL_MODULE__renderers_SubFilterRenderer_js_5cd2c3e9__ from "../renderers/SubFilterRenderer.js";
const FilterListRenderer = (props)=>{
    const { config: configList, variant = __WEBPACK_EXTERNAL_MODULE__types_js_8472748c__.FilterVariants.SIDEBAR, filterApplied = {} } = props;
    const { filters, columnSearch, setSearchValue, setCheckboxValues, setRadioValue, setDateRangeValues, setMonthRangeValues, setSliderValues } = (0, __WEBPACK_EXTERNAL_MODULE__store_js_add3cf67__.useFilterStore)();
    const filterList = (0, __WEBPACK_EXTERNAL_MODULE_react__.useMemo)(()=>{
        const filteredConfigList = columnSearch ? configList.filter((filterItem)=>filterItem.label.toLowerCase().includes(columnSearch.toLowerCase())) : configList;
        return filteredConfigList.map((filterItem)=>{
            const { type, defaultIdentifier, label, filterConfig, collapsed = true, dataId = '' } = filterItem;
            const key = `${type}_${defaultIdentifier}`;
            const hasSubCategories = filterItem.catagories && filterItem.catagories.length > 0;
            let filterComponent = null;
            switch(type){
                case __WEBPACK_EXTERNAL_MODULE__types_js_8472748c__.FilterTypes.SEARCH:
                    filterComponent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__renderers_SearchRenderer_js_43846080__["default"], {
                        key: key,
                        title: label,
                        value: filters[defaultIdentifier]?.value || '',
                        onChange: (value)=>{
                            setSearchValue(defaultIdentifier, value);
                        },
                        placeholder: filterConfig?.defaultValue || 'Search here...',
                        isCollapsed: collapsed,
                        dataId: dataId,
                        identifier: defaultIdentifier
                    });
                    break;
                case __WEBPACK_EXTERNAL_MODULE__types_js_8472748c__.FilterTypes.CHECKBOX:
                    const checkboxOptions = filterConfig?.options || [];
                    filterComponent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__renderers_CheckboxFilterRenderer_js_9adad8cc__["default"], {
                        key: key,
                        title: label,
                        options: checkboxOptions.map((opt)=>({
                                label: opt.label,
                                value: opt.value
                            })),
                        selectedValues: filters[defaultIdentifier]?.values || [],
                        onChange: (values)=>{
                            setCheckboxValues(defaultIdentifier, values);
                        },
                        isCollapsed: collapsed,
                        dataId: dataId,
                        identifier: defaultIdentifier
                    });
                    break;
                case __WEBPACK_EXTERNAL_MODULE__types_js_8472748c__.FilterTypes.DATE_RANGE:
                    filterComponent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__renderers_DateRangeFilterRenderer_js_61fc7460__["default"], {
                        key: key,
                        title: label,
                        min: filters[defaultIdentifier]?.min || '',
                        max: filters[defaultIdentifier]?.max || '',
                        onChange: (min, max)=>{
                            setDateRangeValues(defaultIdentifier, min, max);
                        },
                        isCollapsed: collapsed,
                        dataId: dataId,
                        identifier: defaultIdentifier
                    });
                    break;
                case __WEBPACK_EXTERNAL_MODULE__types_js_8472748c__.FilterTypes.MONTH_RANGE:
                    filterComponent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__renderers_MonthRangeFilterRenderer_js_ecc61406__["default"], {
                        key: key,
                        title: label,
                        min: filters[defaultIdentifier]?.min || '',
                        max: filters[defaultIdentifier]?.max || '',
                        onChange: (min, max)=>{
                            setMonthRangeValues(defaultIdentifier, min, max);
                        },
                        isCollapsed: collapsed,
                        dataId: dataId,
                        identifier: defaultIdentifier
                    });
                    break;
                case __WEBPACK_EXTERNAL_MODULE__types_js_8472748c__.FilterTypes.RADIO:
                    const radioOptions = filterConfig?.options || [];
                    filterComponent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__renderers_RadioFilterRenderer_js_6bf7a591__["default"], {
                        key: key,
                        title: label,
                        options: radioOptions.map((opt)=>({
                                label: opt.label,
                                value: opt.value
                            })),
                        selectedValue: filters[defaultIdentifier]?.value || '',
                        onChange: (value)=>{
                            setRadioValue(defaultIdentifier, value);
                        },
                        isCollapsed: collapsed,
                        dataId: dataId,
                        name: defaultIdentifier,
                        identifier: defaultIdentifier
                    });
                    break;
                case __WEBPACK_EXTERNAL_MODULE__types_js_8472748c__.FilterTypes.RANGE:
                    const { minVal = 0, maxVal = 100 } = filterConfig;
                    const appliedMinVal = filters[defaultIdentifier]?.min !== void 0 ? Number(filters[defaultIdentifier]?.min) : minVal;
                    const appliedMaxVal = filters[defaultIdentifier]?.max !== void 0 ? Number(filters[defaultIdentifier]?.max) : maxVal;
                    filterComponent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__renderers_SliderFilterRenderer_js_3470f8b6__["default"], {
                        key: key,
                        title: label,
                        min: minVal,
                        max: maxVal,
                        minVal: appliedMinVal,
                        maxVal: appliedMaxVal,
                        onChange: (min, max)=>{
                            setSliderValues(defaultIdentifier, min, max);
                        },
                        isCollapsed: collapsed,
                        dataId: dataId,
                        identifier: defaultIdentifier
                    });
                    break;
                default:
                    filterComponent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE_react__["default"].Fragment, null);
            }
            if (hasSubCategories) {
                const FilterWithSubfilter = ()=>{
                    const [isExpanded, setIsExpanded] = (0, __WEBPACK_EXTERNAL_MODULE_react__.useState)(!collapsed);
                    const handleToggle = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((isCollapsed)=>{
                        setIsExpanded(!isCollapsed);
                    }, []);
                    const enhancedFilterComponent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].cloneElement(filterComponent, {
                        onToggle: handleToggle,
                        hasSubFilter: true
                    });
                    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
                        className: "flex flex-col w-full mb-0"
                    }, enhancedFilterComponent, isExpanded && /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__renderers_SubFilterRenderer_js_5cd2c3e9__["default"], {
                        catagories: filterItem.catagories,
                        parentIdentifier: defaultIdentifier
                    }));
                };
                return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(FilterWithSubfilter, {
                    key: key
                });
            }
            return filterComponent;
        });
    }, [
        configList,
        columnSearch,
        filters,
        setSearchValue,
        setCheckboxValues,
        setRadioValue,
        setDateRangeValues,
        setMonthRangeValues,
        setSliderValues
    ]);
    const classNames = (0, __WEBPACK_EXTERNAL_MODULE_classnames__["default"])({
        "block max-h-dropdown overflow-y-scroll overflow-x-auto": variant !== __WEBPACK_EXTERNAL_MODULE__types_js_8472748c__.FilterVariants.SIDEBAR,
        "rounded-t-[var(--radius)] px-5 pt-5 pb-0": variant === __WEBPACK_EXTERNAL_MODULE__types_js_8472748c__.FilterVariants.DROPDOWN_INLINE,
        "flex flex-col w-full mt-0": variant === __WEBPACK_EXTERNAL_MODULE__types_js_8472748c__.FilterVariants.SIDEBAR
    });
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: classNames
    }, filterList);
};
const FilterOptionsCustomRenderers_rslib_entry_ = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].memo(FilterListRenderer);
export { FilterOptionsCustomRenderers_rslib_entry_ as default };
