// @cleartax/truss/dist/components/FilterSidebar/components/FilterSection.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__store_js_add3cf67__ from "../store.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_accordion_js_7c60a3dd__ from "../../ui/accordion.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_button_js_a6e433eb__ from "../../ui/button.js";
const FilterSection = ({ title, isCollapsed = true, children, dataId, identifier, onToggle, hasSubFilter = false })=>{
    const [collapsed, setCollapsed] = (0, __WEBPACK_EXTERNAL_MODULE_react__.useState)(isCollapsed);
    const { filters, resetFilter } = (0, __WEBPACK_EXTERNAL_MODULE__store_js_add3cf67__.useFilterStore)();
    const hasAppliedFilter = identifier && filters[identifier] && (filters[identifier]?.values && (filters[identifier]?.values?.length ?? 0) > 0 || filters[identifier]?.value && filters[identifier]?.value !== '' || filters[identifier]?.min && filters[identifier]?.min !== '' || filters[identifier]?.max && filters[identifier]?.max !== '');
    const getFilterCount = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)(()=>{
        if (!identifier || !filters[identifier]) return 0;
        if (filters[identifier] && filters[identifier]?.values) return filters[identifier]?.values?.length ?? 0;
        return hasAppliedFilter ? 1 : 0;
    }, [
        identifier,
        filters,
        hasAppliedFilter
    ]);
    const filterCount = getFilterCount();
    const handleClearFilter = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((e)=>{
        e.stopPropagation();
        if (identifier) resetFilter(identifier);
    }, [
        identifier,
        resetFilter
    ]);
    const handleToggleAccordion = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((value)=>{
        const newCollapsedState = value !== (identifier || title);
        setCollapsed(newCollapsedState);
        if (onToggle) onToggle(newCollapsedState);
    }, [
        identifier,
        title,
        onToggle
    ]);
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_accordion_js_7c60a3dd__.Accordion, {
        type: "single",
        collapsible: true,
        defaultValue: collapsed ? void 0 : identifier || title,
        onValueChange: handleToggleAccordion,
        className: `${hasSubFilter ? '' : 'border-b'} rounded-none overflow-hidden relative z-1 w-sidebar`,
        "data-id": dataId
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_accordion_js_7c60a3dd__.AccordionItem, {
        value: identifier || title,
        className: "border-0"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_accordion_js_7c60a3dd__.AccordionTrigger, {
        className: "flex justify-between items-center p-3 bg-white cursor-pointer w-full no-underline hover:no-underline [&>svg]:text-primary [&>svg]:transition-transform [&>svg]:duration-300 [&[data-state=open]>svg]:rotate-180 h-10"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex items-center flex-1 min-w-0"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("h4", {
        className: "text-sm font-medium text-sidebar-text-default m-0 truncate max-w-180",
        title: title
    }, title), filterCount > 0 && /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("span", {
        className: "ml-2 w-5 h-5 inline-flex items-center justify-center text-xs bg-primary/10 text-primary rounded-[var(--radius)] font-medium flex-shrink-0"
    }, filterCount)), " ", /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex items-center gap-6"
    }, hasAppliedFilter && /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_button_js_a6e433eb__.Button, {
        variant: "ghost",
        size: "sm",
        onClick: handleClearFilter,
        className: "text-primary text-sm font-medium cursor-pointer hover:bg-transparent hover:text-primary/90 p-0 mr-4 ml-2"
    }, "Clear"))), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_accordion_js_7c60a3dd__.AccordionContent, {
        className: "p-3 bg-white w-full"
    }, children)));
};
const FilterSection_rslib_entry_ = FilterSection;
export { FilterSection_rslib_entry_ as default };
