// @cleartax/truss/dist/components/SelectedFilterV2/FilterCards.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__ui_badge_js_d8455aee__ from "../ui/badge.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_button_js_43a9b5d6__ from "../ui/button.js";
const FilterCards = ({ filter, onClose, closable = true, selectedItem, onClick, dataId, className = '', size = 'sm', dataTestId })=>{
    const { closable: cardClosable = true } = filter;
    const isFilterClosable = closable && cardClosable;
    const onCloseHandler = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)(()=>{
        onClose && onClose(filter.identifier);
    }, [
        filter.identifier,
        onClose
    ]);
    const handleBadgeClick = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)(()=>{
        onClick && onClick(filter.identifier);
    }, [
        filter.identifier,
        onClick
    ]);
    const handleCloseButtonClick = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((e)=>{
        e.stopPropagation();
        onCloseHandler();
    }, [
        onCloseHandler
    ]);
    const isSelected = selectedItem === filter.identifier;
    const badgeSizeClass = {
        sm: "!p-[8px_16px] !text-xs",
        md: "!p-[10px_20px] !text-sm",
        lg: "!p-[12px_24px] !text-base"
    }[size];
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_badge_js_d8455aee__.Badge, {
        className: `${className} ${badgeSizeClass}`,
        "data-testid": dataTestId,
        variant: isSelected ? 'secondary' : 'outline',
        onClick: handleBadgeClick,
        "data-id": dataId ? `${dataId}-CONTAINER` : void 0
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex items-center gap-2 text-filter-text",
        "data-id": dataId ? `${dataId}-TEXT` : void 0
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("span", {
        className: "font-medium"
    }, filter.name), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("span", null, ":"), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("span", null, filter.value), isFilterClosable && /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_button_js_43a9b5d6__.Button, {
        variant: "ghost",
        size: "sm",
        onClick: handleCloseButtonClick,
        "aria-label": `Remove ${filter.name} filter`
    })));
};
const FilterCards_rslib_entry_ = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].memo(FilterCards);
export { FilterCards_rslib_entry_ as default };
