// @cleartax/truss/dist/components/SelectedFilterV2/index.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__FilterSidebar_store_js_3c57d29f__ from "../FilterSidebar/store.js";
import * as __WEBPACK_EXTERNAL_MODULE__FilterCards_js_f5e4a4fe__ from "./FilterCards.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_button_js_43a9b5d6__ from "../ui/button.js";
const SelectedFilterV2 = ({ className = '', size = 'md', closable = true, filtersApplied = [], showClearAll = false, onClearAllClick, onFilterRemove, onClick, selectedItem, dataId, selectedFilterTextConfig })=>{
    const containerRef = (0, __WEBPACK_EXTERNAL_MODULE_react__.useRef)(null);
    const handleClearAllClick = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)(()=>{
        const { resetFilters } = __WEBPACK_EXTERNAL_MODULE__FilterSidebar_store_js_3c57d29f__.useFilterStore.getState();
        if (onClearAllClick) {
            onClearAllClick();
            resetFilters();
        }
    }, [
        onClearAllClick
    ]);
    const filterCardsList = (0, __WEBPACK_EXTERNAL_MODULE_react__.useMemo)(()=>filtersApplied.map((filter, index)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__FilterCards_js_f5e4a4fe__["default"], {
                key: `${filter.identifier}-${index}`,
                filter: filter,
                size: size,
                onClose: onFilterRemove,
                closable: closable,
                onClick: onClick,
                selectedItem: selectedItem,
                dataId: dataId
            })), [
        filtersApplied,
        size,
        onFilterRemove,
        closable,
        onClick,
        selectedItem,
        dataId
    ]);
    if (!filtersApplied.length) return null;
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: `p-2 w-full ${className}`,
        ref: containerRef
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "inline-flex flex-wrap gap-2 items-baseline w-full"
    }, filterCardsList, showClearAll && filtersApplied.length > 0 && /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_button_js_43a9b5d6__.Button, {
        variant: "ghost",
        size: "sm",
        onClick: handleClearAllClick,
        "aria-label": "Clear all filters",
        className: "text-filter-text text-xs h-auto hover:bg-transparent hover:text-filter-text ml-0"
    }, selectedFilterTextConfig?.clearAll || 'Clear All')));
};
const SelectedFiltersV2 = SelectedFilterV2;
export { SelectedFilterV2, SelectedFiltersV2 };
