// @cleartax/truss/dist/components/FilterSidebar/index.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__store_js_535ade31__ from "./store.js";
import * as __WEBPACK_EXTERNAL_MODULE__components_SidebarFooter_js_b7b47da5__ from "./components/SidebarFooter.js";
import * as __WEBPACK_EXTERNAL_MODULE__components_FilterOptionsCustomRenderers_js_21067c5f__ from "./components/FilterOptionsCustomRenderers.js";
import * as __WEBPACK_EXTERNAL_MODULE__renderers_ColumnSearchRenderer_js_28bdd177__ from "./renderers/ColumnSearchRenderer.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_button_js_43a9b5d6__ from "../ui/button.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_scroll_area_js_4d32fdc5__ from "../ui/scroll-area.js";
const FilterSidebar = ({ onClose, onApply, filterConfig })=>{
    const { resetFilters, columnSearch, setColumnSearch } = (0, __WEBPACK_EXTERNAL_MODULE__store_js_535ade31__.useFilterStore)();
    const { getFilters } = (0, __WEBPACK_EXTERNAL_MODULE__store_js_535ade31__.useFilterStore)();
    const handleReset = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)(()=>{
        resetFilters();
        setColumnSearch('');
    }, [
        resetFilters,
        setColumnSearch
    ]);
    const handleApply = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)(()=>{
        onApply(getFilters());
    }, [
        onApply,
        getFilters
    ]);
    const handleColumnSearchChange = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((value)=>{
        setColumnSearch(value);
    }, [
        setColumnSearch
    ]);
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex flex-col relative h-screen bg-white overflow-hidden w-sidebar border-r border-r-border-color"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex sticky top-0 flex-row justify-between items-center bg-white p-4"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("h4", {
        className: "text-lg font-medium"
    }, "Filters"), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_button_js_43a9b5d6__.Button, {
        variant: "ghost",
        size: "icon",
        className: "cursor-pointer text-gray-500 hover:text-gray-700 text-xl flex items-center justify-center",
        onClick: onClose
    }, "\xd7")), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__renderers_ColumnSearchRenderer_js_28bdd177__["default"], {
        value: columnSearch,
        onChange: handleColumnSearchChange,
        placeholder: "Search for Filters"
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_scroll_area_js_4d32fdc5__.ScrollArea, {
        className: "flex-1 overflow-y-auto pb-[76px]"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__components_FilterOptionsCustomRenderers_js_21067c5f__["default"], {
        config: filterConfig
    })), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__components_SidebarFooter_js_b7b47da5__["default"], {
        handleReset: handleReset,
        handleApply: handleApply
    }));
};
const FilterSidebar_rslib_entry_ = FilterSidebar;
export { FilterSidebar_rslib_entry_ as default };
