// @cleartax/truss/dist/components/FilterSidebar/store.js
import * as __WEBPACK_EXTERNAL_MODULE_zustand__ from "zustand";
import * as __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__ from "./types.js";
const useFilterStore = (0, __WEBPACK_EXTERNAL_MODULE_zustand__.create)((set, get)=>({
        filters: {},
        columnSearch: '',
        setFilter: (identifier, type, value, fieldName)=>{
            set((state)=>({
                    filters: {
                        ...state.filters,
                        [identifier]: {
                            filterType: type,
                            value: type === __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.SEARCH || type === __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.RADIO ? value : void 0,
                            values: type === __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.CHECKBOX ? value : void 0,
                            min: (type === __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.DATE_RANGE || type === __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.MONTH_RANGE) && value.min ? value.min : void 0,
                            max: (type === __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.DATE_RANGE || type === __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.MONTH_RANGE) && value.max ? value.max : void 0,
                            source: fieldName || identifier
                        }
                    }
                }));
        },
        setSearchValue: (identifier, value, fieldName)=>{
            set((state)=>({
                    filters: {
                        ...state.filters,
                        [identifier]: {
                            filterType: __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.SEARCH,
                            value,
                            source: fieldName || identifier
                        }
                    }
                }));
        },
        setCheckboxValues: (identifier, values, fieldName)=>{
            set((state)=>({
                    filters: {
                        ...state.filters,
                        [identifier]: {
                            filterType: __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.CHECKBOX,
                            values,
                            source: fieldName || identifier
                        }
                    }
                }));
        },
        setRadioValue: (identifier, value, fieldName)=>{
            set((state)=>({
                    filters: {
                        ...state.filters,
                        [identifier]: {
                            filterType: __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.RADIO,
                            value,
                            source: fieldName || identifier
                        }
                    }
                }));
        },
        setDateRangeValues: (identifier, min, max, fieldName)=>{
            set((state)=>({
                    filters: {
                        ...state.filters,
                        [identifier]: {
                            filterType: __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.DATE_RANGE,
                            min,
                            max,
                            source: fieldName || identifier
                        }
                    }
                }));
        },
        setMonthRangeValues: (identifier, min, max, fieldName)=>{
            set((state)=>({
                    filters: {
                        ...state.filters,
                        [identifier]: {
                            filterType: __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.MONTH_RANGE,
                            min,
                            max,
                            source: fieldName || identifier
                        }
                    }
                }));
        },
        setCatagories: (identifier, catagories)=>{
            set((state)=>{
                const currentFilter = state.filters[identifier] || {};
                return {
                    filters: {
                        ...state.filters,
                        [identifier]: {
                            ...currentFilter,
                            catagories
                        }
                    }
                };
            });
        },
        setColumnSearch: (value)=>{
            set({
                columnSearch: value
            });
        },
        getFilters: ()=>{
            const { filters } = get();
            return {
                ...filters
            };
        },
        resetFilters: ()=>{
            set({
                filters: {}
            });
        },
        resetFilter: (identifier)=>{
            set((state)=>{
                const newFilters = {
                    ...state.filters
                };
                delete newFilters[identifier];
                return {
                    filters: newFilters
                };
            });
        },
        setDateValue: (identifier, value, fieldName)=>{
            set((state)=>({
                    filters: {
                        ...state.filters,
                        [identifier]: {
                            filterType: __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.DATE,
                            value,
                            source: fieldName || identifier
                        }
                    }
                }));
        },
        setMonthValue: (identifier, value, fieldName)=>{
            set((state)=>({
                    filters: {
                        ...state.filters,
                        [identifier]: {
                            filterType: __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.MONTH,
                            value,
                            source: fieldName || identifier
                        }
                    }
                }));
        },
        setSliderValues: (identifier, min, max, fieldName)=>{
            set((state)=>({
                    filters: {
                        ...state.filters,
                        [identifier]: {
                            filterType: __WEBPACK_EXTERNAL_MODULE__types_js_7eec16d2__.FilterTypes.RANGE,
                            min,
                            max,
                            source: fieldName || identifier
                        }
                    }
                }));
        }
    }));
export { useFilterStore };
