// @cleartax/truss/dist/components/FilterSidebar/types.js
import "react";
var types_rslib_entry_FilterTypes = /*#__PURE__*/ function(FilterTypes) {
    FilterTypes["SEARCH"] = "search";
    FilterTypes["CHECKBOX"] = "checkbox";
    FilterTypes["RANGE"] = "range";
    FilterTypes["DATE_RANGE"] = "dateRange";
    FilterTypes["MONTH_RANGE"] = "monthRange";
    FilterTypes["DATE"] = "date";
    FilterTypes["MONTH"] = "month";
    FilterTypes["SEARCH_AND_SELECT"] = "searchnselect";
    FilterTypes["RADIO"] = "radio";
    return FilterTypes;
}({});
var types_rslib_entry_FilterVariants = /*#__PURE__*/ function(FilterVariants) {
    FilterVariants["SIDEBAR"] = "SIDEBAR";
    FilterVariants["DROPDOWN"] = "DROPDOWN";
    FilterVariants["DROPDOWN_INLINE"] = "DROPDOWN_INLINE";
    return FilterVariants;
}({});
var types_rslib_entry_SideBarFilterVariants = /*#__PURE__*/ function(SideBarFilterVariants) {
    SideBarFilterVariants["APPLY_BUTTON"] = "APPLY_BUTTON";
    SideBarFilterVariants["NO_APPLY_BUTTON"] = "NO_APPLY_BUTTON";
    return SideBarFilterVariants;
}({});
var types_rslib_entry_DATE_FORMATS = /*#__PURE__*/ function(DATE_FORMATS) {
    DATE_FORMATS["ISO"] = "ISO";
    DATE_FORMATS["MMYYYY"] = "MMYYYY";
    DATE_FORMATS["YYYYMMDD"] = "YYYY-MM-DD";
    DATE_FORMATS["DDMMYY"] = "DD/MM/YY";
    DATE_FORMATS["DDMMYYYY"] = "DD-MMM-YYYY";
    return DATE_FORMATS;
}({});
var types_rslib_entry_FILTER_ACTIONS = /*#__PURE__*/ function(FILTER_ACTIONS) {
    FILTER_ACTIONS["REMOVE_FILTER"] = "REMOVE_FILTER";
    FILTER_ACTIONS["RESET_FILTER"] = "RESET_FILTER";
    FILTER_ACTIONS["APPLY_FILTER"] = "APPLY_FILTER";
    FILTER_ACTIONS["SHOW_APPLIED_FILTER"] = "SHOW_APPLIED_FILTER";
    FILTER_ACTIONS["SHOW_ALL_FILTER"] = "SHOW_ALL_FILTER";
    FILTER_ACTIONS["INITIALIZE_FILTER"] = "INITIALIZE_FILTER";
    return FILTER_ACTIONS;
}({});
var types_rslib_entry_SIDEBAR_FILTER_ACTIONS = /*#__PURE__*/ function(SIDEBAR_FILTER_ACTIONS) {
    SIDEBAR_FILTER_ACTIONS["SIDEBAR_SINGLE_APPLY_FILTER"] = "SIDEBAR_SINGLE_APPLY_FILTER";
    SIDEBAR_FILTER_ACTIONS["SIDEBAR_SINGLE_REMOVE_FILTER"] = "SIDEBAR_SINGLE_REMOVE_FILTER";
    return SIDEBAR_FILTER_ACTIONS;
}({});
var types_rslib_entry_DELIMITERS = /*#__PURE__*/ function(DELIMITERS) {
    DELIMITERS["SPACE"] = " ";
    DELIMITERS["PIPE"] = "|";
    return DELIMITERS;
}({});
var types_rslib_entry_DrawerStates = /*#__PURE__*/ function(DrawerStates) {
    DrawerStates["Open"] = "OPEN";
    DrawerStates["Closed"] = "CLOSED";
    DrawerStates["Collapsed"] = "COLLAPSED";
    return DrawerStates;
}({});
export { types_rslib_entry_DATE_FORMATS as DATE_FORMATS, types_rslib_entry_DELIMITERS as DELIMITERS, types_rslib_entry_DrawerStates as DrawerStates, types_rslib_entry_FILTER_ACTIONS as FILTER_ACTIONS, types_rslib_entry_FilterTypes as FilterTypes, types_rslib_entry_FilterVariants as FilterVariants, types_rslib_entry_SIDEBAR_FILTER_ACTIONS as SIDEBAR_FILTER_ACTIONS, types_rslib_entry_SideBarFilterVariants as SideBarFilterVariants };
