// @cleartax/truss/dist/components/ui/molecule/date-range-picker.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE_date_fns_925206bc__ from "date-fns";
import * as __WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__ from "lucide-react";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__ from "../../../lib/utils.js";
import * as __WEBPACK_EXTERNAL_MODULE__button_js_f7141515__ from "../button.js";
import * as __WEBPACK_EXTERNAL_MODULE__calendar_v2_js_11723ab3__ from "../calendar-v2.js";
import * as __WEBPACK_EXTERNAL_MODULE__popover_js_624da78e__ from "../popover.js";
function DateRangePicker({ startDate, endDate, onStartDateChange, onEndDateChange, activeInput, onActiveInputChange, disabled, className }) {
    const formatDisplayDate = (date)=>{
        if (!date) return "DD/MM/YY";
        return (0, __WEBPACK_EXTERNAL_MODULE_date_fns_925206bc__.format)(date, "dd/MM/yy");
    };
    const handleDateSelect = __WEBPACK_EXTERNAL_MODULE_react__.useCallback((date)=>{
        if (!date) return;
        if ('start' === activeInput) {
            if (endDate && date > endDate) onEndDateChange(void 0);
            onStartDateChange(date);
            onActiveInputChange('end');
        } else if ('end' === activeInput) {
            if (startDate && date < startDate) {
                onStartDateChange(date);
                onEndDateChange(void 0);
            } else {
                onEndDateChange(date);
                onActiveInputChange(null);
            }
        }
    }, [
        activeInput,
        startDate,
        endDate,
        onStartDateChange,
        onEndDateChange,
        onActiveInputChange
    ]);
    const handlePopoverOpenChange = __WEBPACK_EXTERNAL_MODULE_react__.useCallback((open)=>{
        if (!open) onActiveInputChange(null);
    }, [
        onActiveInputChange
    ]);
    const handleStartDateButtonClick = __WEBPACK_EXTERNAL_MODULE_react__.useCallback(()=>onActiveInputChange('start'), [
        onActiveInputChange
    ]);
    const handleEndDateButtonClick = __WEBPACK_EXTERNAL_MODULE_react__.useCallback(()=>onActiveInputChange('end'), [
        onActiveInputChange
    ]);
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__popover_js_624da78e__.Popover, {
        open: null !== activeInput,
        onOpenChange: handlePopoverOpenChange
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__popover_js_624da78e__.PopoverTrigger, {
        asChild: true
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__.cn)("flex items-center justify-center", className)
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__button_js_f7141515__.Button, {
        variant: "outline",
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__.cn)("w-100 justify-start text-left font-normal h-10 px-2", startDate ? "text-foreground" : "text-muted-foreground", 'start' === activeInput ? "border-[hsl(var(--primary))] bg-primary/5 font-medium" : " "),
        onClick: handleStartDateButtonClick
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.CalendarIcon, {
        className: "mr-1 h-3.5 w-3.5 flex-shrink-0"
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", {
        className: "text-xs truncate"
    }, formatDisplayDate(startDate))), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", {
        className: "mx-2 text-base font-bold"
    }, "→"), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__button_js_f7141515__.Button, {
        variant: "outline",
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__.cn)("w-100 justify-start text-left font-normal h-10 px-2", endDate ? "text-foreground" : "text-muted-foreground", 'end' === activeInput ? "border-[hsl(var(--primary))] bg-primary/5 font-medium" : " "),
        onClick: handleEndDateButtonClick
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.CalendarIcon, {
        className: "mr-1 h-3.5 w-3.5 flex-shrink-0"
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", {
        className: "text-xs truncate"
    }, formatDisplayDate(endDate))))), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__popover_js_624da78e__.PopoverContent, {
        className: "w-auto p-0",
        align: "center"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__calendar_v2_js_11723ab3__.CalendarV2, {
        mode: "single",
        selected: 'start' === activeInput ? startDate : endDate,
        onSelect: handleDateSelect,
        disabled: disabled,
        initialFocus: true
    })));
}
DateRangePicker.displayName = "DateRangePicker";
export { DateRangePicker };
