// @cleartax/truss/dist/components/ui/molecule/month-range-picker.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__ from "lucide-react";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__ from "../../../lib/utils.js";
import * as __WEBPACK_EXTERNAL_MODULE__button_js_f7141515__ from "../button.js";
import * as __WEBPACK_EXTERNAL_MODULE__monthSelector_js_e4e4d62f__ from "../monthSelector.js";
import * as __WEBPACK_EXTERNAL_MODULE__yearSelector_js_d352f730__ from "../yearSelector.js";
import * as __WEBPACK_EXTERNAL_MODULE__popover_js_624da78e__ from "../popover.js";
import * as __WEBPACK_EXTERNAL_MODULE__hooks_useMonthRangePicker_js_e359c505__ from "../../../hooks/useMonthRangePicker.js";
function MonthRangePicker({ fromDate, toDate, onFromDateChange, onToDateChange, activeInput, onActiveInputChange, className }) {
    const { selectorType, yearRange, displayFromDate, displayToDate, selectedMonthForSelector, handleMonthSelect, handleYearSelect, handlePreviousYearRange, handleNextYearRange, handleYearClick, handleSelectedMonthChange, handlePopoverOpenChange, handleFromButtonClick, handleToButtonClick } = (0, __WEBPACK_EXTERNAL_MODULE__hooks_useMonthRangePicker_js_e359c505__.useMonthRangePicker)({
        fromDate,
        toDate,
        onFromDateChange,
        onToDateChange,
        activeInput,
        onActiveInputChange
    });
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__popover_js_624da78e__.Popover, {
        open: null !== activeInput,
        onOpenChange: handlePopoverOpenChange
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__popover_js_624da78e__.PopoverTrigger, {
        asChild: true
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__.cn)("flex items-center justify-center", className)
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__button_js_f7141515__.Button, {
        variant: "outline",
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__.cn)("w-100 justify-start text-left font-normal h-10 px-2", fromDate ? "text-foreground" : "text-muted-foreground", 'from' === activeInput ? "border-[hsl(var(--primary))] bg-primary/5 font-medium" : " "),
        onClick: handleFromButtonClick
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.CalendarIcon, {
        className: "mr-1 h-3.5 w-3.5 flex-shrink-0"
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", {
        className: "text-xs truncate"
    }, displayFromDate)), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", {
        className: "mx-2 text-base font-bold"
    }, "→"), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__button_js_f7141515__.Button, {
        variant: "outline",
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__.cn)("w-100 justify-start text-left font-normal h-10 px-2", toDate ? "text-foreground" : "text-muted-foreground", 'to' === activeInput ? "border-[hsl(var(--primary))] bg-primary/5 font-medium" : " "),
        onClick: handleToButtonClick
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.CalendarIcon, {
        className: "mr-1 h-3.5 w-3.5 flex-shrink-0"
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", {
        className: "text-xs truncate"
    }, displayToDate)))), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__popover_js_624da78e__.PopoverContent, {
        className: "w-auto p-0",
        align: "center"
    }, 'month' === selectorType ? /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__monthSelector_js_e4e4d62f__.MonthSelector, {
        selectedMonth: selectedMonthForSelector,
        onMonthSelect: handleMonthSelect,
        onYearClick: handleYearClick,
        onSelectedMonthChange: handleSelectedMonthChange
    }) : /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__yearSelector_js_d352f730__.YearSelector, {
        selectedMonth: selectedMonthForSelector,
        yearRange: yearRange,
        onYearSelect: handleYearSelect,
        onPreviousYearRange: handlePreviousYearRange,
        onNextYearRange: handleNextYearRange
    })));
}
MonthRangePicker.displayName = "MonthRangePicker";
export { MonthRangePicker };
