// @cleartax/truss/dist/components/FilterSidebar/renderers/CheckboxFilterRenderer.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__ from "../components/FilterSection.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_checkbox_js_fbbac507__ from "../../ui/checkbox.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_label_js_64ca4bc7__ from "../../ui/label.js";
const CheckboxFilterRenderer = ({ title, options, selectedValues, onChange, isCollapsed = false, dataId, identifier, onToggle })=>{
    const handleCheckboxChange = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((value, checked)=>{
        checked ? onChange([
            ...selectedValues,
            value
        ]) : onChange(selectedValues.filter((v)=>v !== value));
    }, [
        onChange,
        selectedValues
    ]);
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__["default"], {
        title: title,
        isCollapsed: isCollapsed,
        dataId: dataId,
        identifier: identifier,
        onToggle: onToggle
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex flex-col gap-2 max-h-filter-section overflow-y-auto w-full"
    }, options.map((option)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
            key: option.value,
            className: "flex flex-row items-center gap-2 w-full"
        }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_checkbox_js_fbbac507__.Checkbox, {
            id: `${identifier}-${option.value}`,
            checked: selectedValues.includes(option.value),
            onCheckedChange: (checked)=>handleCheckboxChange(option.value, checked)
        }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_label_js_64ca4bc7__.Label, {
            htmlFor: `${identifier}-${option.value}`,
            className: "flex items-center text-sm text-sidebar-text-default cursor-pointer ml-2"
        }, option.label)))));
};
const CheckboxFilterRenderer_rslib_entry_ = CheckboxFilterRenderer;
export { CheckboxFilterRenderer_rslib_entry_ as default };
