// @cleartax/truss/dist/components/FilterSidebar/renderers/DateFilterRenderer.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE_date_fns_925206bc__ from "date-fns";
import * as __WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__ from "lucide-react";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__ from "../../../lib/utils.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_button_js_a6e433eb__ from "../../ui/button.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_calendar_v2_js_e71266d0__ from "../../ui/calendar-v2.js";
import * as __WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__ from "../components/FilterSection.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_popover_js_fe97bff3__ from "../../ui/popover.js";
const DateFilterRenderer = ({ title, value, onChange, isCollapsed = false, dataId, identifier, onToggle })=>{
    const [selectedDate, setSelectedDate] = (0, __WEBPACK_EXTERNAL_MODULE_react__.useState)(value ? new Date(value) : void 0);
    const [isOpen, setIsOpen] = (0, __WEBPACK_EXTERNAL_MODULE_react__.useState)(false);
    (0, __WEBPACK_EXTERNAL_MODULE_react__.useEffect)(()=>{
        setSelectedDate(value ? new Date(value) : void 0);
    }, [
        value
    ]);
    const formatDisplayDate = (date)=>{
        if (!date) return 'DD/MM/YY';
        return (0, __WEBPACK_EXTERNAL_MODULE_date_fns_925206bc__.format)(date, "dd/MM/yy");
    };
    const handleDateSelect = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((date)=>{
        if (!date) return;
        setSelectedDate(date);
        setIsOpen(false);
        const formattedDate = (0, __WEBPACK_EXTERNAL_MODULE_date_fns_925206bc__.format)(date, "yyyy-MM-dd");
        onChange(formattedDate);
    }, [
        onChange,
        setSelectedDate,
        setIsOpen
    ]);
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__["default"], {
        title: title,
        isCollapsed: isCollapsed,
        dataId: dataId,
        identifier: identifier,
        onToggle: onToggle
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex flex-col w-full px-0"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_popover_js_fe97bff3__.Popover, {
        open: isOpen,
        onOpenChange: setIsOpen
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_popover_js_fe97bff3__.PopoverTrigger, {
        asChild: true
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_button_js_a6e433eb__.Button, {
        variant: "outline",
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__.cn)("w-full justify-start text-left font-normal h-10", selectedDate ? "text-foreground" : "text-muted-foreground", isOpen ? "border-[hsl(var(--primary))] bg-primary/5 font-medium" : "")
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.CalendarIcon, {
        className: "mr-2 h-4 w-4"
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("span", null, formatDisplayDate(selectedDate)))), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_popover_js_fe97bff3__.PopoverContent, {
        className: "w-auto p-0",
        align: "center"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_calendar_v2_js_e71266d0__.CalendarV2, {
        mode: "single",
        selected: selectedDate,
        onSelect: handleDateSelect,
        disabled: (date)=>date > new Date(),
        initialFocus: true
    })))));
};
const DateFilterRenderer_rslib_entry_ = DateFilterRenderer;
export { DateFilterRenderer_rslib_entry_ as default };
