// @cleartax/truss/dist/components/FilterSidebar/renderers/DateRangeFilterRenderer.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE_moment__ from "moment";
import * as __WEBPACK_EXTERNAL_MODULE__ui_molecule_date_range_picker_js_f1be0668__ from "../../ui/molecule/date-range-picker.js";
import * as __WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__ from "../components/FilterSection.js";
const DateRangeFilterRenderer = ({ title, min, max, onChange, isCollapsed = false, dataId, identifier, onToggle })=>{
    const [activeInput, setActiveInput] = (0, __WEBPACK_EXTERNAL_MODULE_react__.useState)(null);
    const startDateFromProps = min ? new Date(min) : void 0;
    const endDateFromProps = max ? new Date(max) : void 0;
    const contentRef = (0, __WEBPACK_EXTERNAL_MODULE_react__.useRef)(null);
    const handleStartDateChange = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((date)=>{
        onChange(date ? (0, __WEBPACK_EXTERNAL_MODULE_moment__["default"])(date).format('YYYY-MM-DD') : '', max || '');
    }, [
        onChange,
        max
    ]);
    const handleEndDateChange = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((date)=>{
        onChange(min || '', date ? (0, __WEBPACK_EXTERNAL_MODULE_moment__["default"])(date).format('YYYY-MM-DD') : '');
    }, [
        onChange,
        min
    ]);
    const handleActiveInputChange = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((input)=>{
        setActiveInput(input);
    }, []);
    const handleDisabledDate = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((date)=>date > new Date(), []);
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__["default"], {
        title: title,
        isCollapsed: isCollapsed,
        dataId: dataId,
        identifier: identifier,
        onToggle: onToggle
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        ref: contentRef,
        style: {
            transition: 'height 0.3s ease',
            overflow: 'visible'
        }
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex flex-col w-full px-0"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_molecule_date_range_picker_js_f1be0668__.DateRangePicker, {
        startDate: startDateFromProps,
        endDate: endDateFromProps,
        onStartDateChange: handleStartDateChange,
        onEndDateChange: handleEndDateChange,
        activeInput: activeInput,
        onActiveInputChange: handleActiveInputChange,
        disabled: handleDisabledDate,
        className: "mb-4"
    }))));
};
const DateRangeFilterRenderer_rslib_entry_ = DateRangeFilterRenderer;
export { DateRangeFilterRenderer_rslib_entry_ as default };
