// @cleartax/truss/dist/components/FilterSidebar/renderers/MonthRangeFilterRenderer.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE_moment__ from "moment";
import * as __WEBPACK_EXTERNAL_MODULE__ui_molecule_month_range_picker_js_f0d1c57f__ from "../../ui/molecule/month-range-picker.js";
import * as __WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__ from "../components/FilterSection.js";
const MonthRangeFilterRenderer = ({ title, min, max, onChange, isCollapsed = false, dataId, identifier, onToggle })=>{
    const [activeInput, setActiveInput] = (0, __WEBPACK_EXTERNAL_MODULE_react__.useState)(null);
    const fromDateFromProps = min ? new Date(min) : void 0;
    const toDateFromProps = max ? new Date(max) : void 0;
    const contentRef = (0, __WEBPACK_EXTERNAL_MODULE_react__.useRef)(null);
    const handleFromDateChange = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((date)=>{
        onChange(date ? (0, __WEBPACK_EXTERNAL_MODULE_moment__["default"])(date).format('YYYY-MM') : '', max || '');
    }, [
        onChange,
        max
    ]);
    const handleToDateChange = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((date)=>{
        onChange(min || '', date ? (0, __WEBPACK_EXTERNAL_MODULE_moment__["default"])(date).format('YYYY-MM') : '');
    }, [
        onChange,
        min
    ]);
    const handleActiveInputChange = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((input)=>{
        setActiveInput(input);
    }, []);
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__["default"], {
        title: title,
        isCollapsed: isCollapsed,
        dataId: dataId,
        identifier: identifier,
        onToggle: onToggle
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        ref: contentRef,
        style: {
            transition: 'height 0.3s ease',
            overflow: 'visible'
        }
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex flex-col w-full px-0"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_molecule_month_range_picker_js_f0d1c57f__.MonthRangePicker, {
        fromDate: fromDateFromProps,
        toDate: toDateFromProps,
        onFromDateChange: handleFromDateChange,
        onToDateChange: handleToDateChange,
        activeInput: activeInput,
        onActiveInputChange: handleActiveInputChange,
        className: "mb-4"
    }))));
};
const MonthRangeFilterRenderer_rslib_entry_ = MonthRangeFilterRenderer;
export { MonthRangeFilterRenderer_rslib_entry_ as default };
