// @cleartax/truss/dist/components/FilterSidebar/renderers/MonthFilterRenderer.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__ from "lucide-react";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__ from "../../../lib/utils.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_button_js_a6e433eb__ from "../../ui/button.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_monthSelector_js_711aae35__ from "../../ui/monthSelector.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_yearSelector_js_1e01f0dd__ from "../../ui/yearSelector.js";
import * as __WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__ from "../components/FilterSection.js";
import * as __WEBPACK_EXTERNAL_MODULE__hooks_useMonthFilter_js_aeaff02d__ from "../../../hooks/useMonthFilter.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_popover_js_fe97bff3__ from "../../ui/popover.js";
const MonthFilterRenderer = ({ title, value, onChange, isCollapsed = false, dataId, identifier, onToggle })=>{
    const { isOpen, selectedDate, selectorType, yearRange, displayDate, handlePopoverOpenChange, handleMonthSelect, handleYearSelect, handlePreviousYearRange, handleNextYearRange, handleYearClick, handleSelectedMonthChange } = (0, __WEBPACK_EXTERNAL_MODULE__hooks_useMonthFilter_js_aeaff02d__.useMonthFilter)({
        value,
        onChange
    });
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__["default"], {
        title: title,
        isCollapsed: isCollapsed,
        dataId: dataId,
        identifier: identifier,
        onToggle: onToggle
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex flex-col w-full px-0"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_popover_js_fe97bff3__.Popover, {
        open: isOpen,
        onOpenChange: handlePopoverOpenChange
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_popover_js_fe97bff3__.PopoverTrigger, {
        asChild: true
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_button_js_a6e433eb__.Button, {
        variant: "outline",
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_d0b7f1af__.cn)("w-full justify-start text-left font-normal h-10", value ? "text-foreground" : "text-muted-foreground", isOpen ? "border-[hsl(var(--primary))] bg-primary/5 font-medium" : "")
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.CalendarIcon, {
        className: "mr-2 h-4 w-4"
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("span", null, displayDate))), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_popover_js_fe97bff3__.PopoverContent, {
        className: "w-auto p-0",
        align: "center"
    }, 'month' === selectorType ? /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_monthSelector_js_711aae35__.MonthSelector, {
        selectedMonth: selectedDate,
        onMonthSelect: handleMonthSelect,
        onYearClick: handleYearClick,
        onSelectedMonthChange: handleSelectedMonthChange
    }) : /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_yearSelector_js_1e01f0dd__.YearSelector, {
        selectedMonth: selectedDate,
        yearRange: yearRange,
        onYearSelect: handleYearSelect,
        onPreviousYearRange: handlePreviousYearRange,
        onNextYearRange: handleNextYearRange
    })))));
};
const MonthFilterRenderer_rslib_entry_ = MonthFilterRenderer;
export { MonthFilterRenderer_rslib_entry_ as default };
