// @cleartax/truss/dist/components/FilterSidebar/renderers/RadioFilterRenderer.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__ from "../components/FilterSection.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_radio_group_js_37cb45e4__ from "../../ui/radio-group.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_label_js_64ca4bc7__ from "../../ui/label.js";
const RadioFilterRenderer = ({ title, options, selectedValue, onChange, isCollapsed = false, dataId, name, identifier, onToggle })=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__["default"], {
        title: title,
        isCollapsed: isCollapsed,
        dataId: dataId,
        identifier: identifier,
        onToggle: onToggle
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_radio_group_js_37cb45e4__.RadioGroup, {
        value: selectedValue,
        onValueChange: onChange,
        className: "flex flex-col gap-2 w-full"
    }, options.map((option)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
            key: option.value,
            className: "flex flex-row items-center gap-2 w-full py-1"
        }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_radio_group_js_37cb45e4__.RadioGroupItem, {
            value: option.value,
            id: `${name}-${option.value}`
        }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_label_js_64ca4bc7__.Label, {
            htmlFor: `${name}-${option.value}`,
            className: "flex items-center text-sm text-sidebar-text-default cursor-pointer ml-2"
        }, option.label)))));
const RadioFilterRenderer_rslib_entry_ = RadioFilterRenderer;
export { RadioFilterRenderer_rslib_entry_ as default };
