// @cleartax/truss/dist/components/FilterSidebar/renderers/SearchRenderer.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__ from "../components/FilterSection.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_input_js_dfab719b__ from "../../ui/input.js";
const SearchRenderer = ({ title, value, onChange, placeholder = 'Search here...', isCollapsed = false, dataId, identifier, onToggle })=>{
    const handleInputChange = (0, __WEBPACK_EXTERNAL_MODULE_react__.useCallback)((e)=>{
        onChange(e.target.value);
    }, [
        onChange
    ]);
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__["default"], {
        title: title,
        isCollapsed: isCollapsed,
        dataId: dataId,
        identifier: identifier,
        onToggle: onToggle
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "relative w-full"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_input_js_dfab719b__.Input, {
        type: "text",
        className: "w-full py-2 px-3 pl-10 border text-sm outline-none transition-all duration-200 relative focus:border focus:shadow-input-focus",
        placeholder: placeholder,
        value: value,
        onChange: handleInputChange
    })));
};
const SearchRenderer_rslib_entry_ = SearchRenderer;
export { SearchRenderer_rslib_entry_ as default };
