// @cleartax/truss/dist/components/FilterSidebar/renderers/SliderFilterRenderer.js
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__ from "../components/FilterSection.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_sliderV2_js_b4599b3e__ from "../../ui/sliderV2.js";
import * as __WEBPACK_EXTERNAL_MODULE__ui_input_js_dfab719b__ from "../../ui/input.js";
import * as __WEBPACK_EXTERNAL_MODULE__hooks_useSliderFilter_js_9b6a8321__ from "../../../hooks/useSliderFilter.js";
const SliderFilterRenderer = ({ title, min, max, minVal, maxVal, onChange, isCollapsed = false, dataId, identifier, onToggle })=>{
    const { values, inputMin, inputMax, isFilterApplied, safeMin, safeMax, handleSliderChange, handleInputMinChange, handleInputMaxChange, handleInputMinBlur, handleInputMaxBlur, handleMinKeyDown, handleMaxKeyDown } = (0, __WEBPACK_EXTERNAL_MODULE__hooks_useSliderFilter_js_9b6a8321__.useSliderFilter)({
        min,
        max,
        minVal,
        maxVal,
        onChange
    });
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__components_FilterSection_js_5960f07f__["default"], {
        title: title,
        isCollapsed: isCollapsed,
        dataId: dataId,
        identifier: identifier,
        onToggle: onToggle
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex flex-col w-full px-0 py-4"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "mb-4"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_sliderV2_js_b4599b3e__.SliderV2, {
        defaultValue: values,
        min: safeMin,
        max: safeMax,
        step: 1,
        value: values,
        onValueChange: handleSliderChange,
        className: `w-full ${isFilterApplied ? 'filter-applied' : ''}`
    })), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("div", {
        className: "flex items-center justify-between"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_input_js_dfab719b__.Input, {
        type: "text",
        value: inputMin,
        onChange: handleInputMinChange,
        onBlur: handleInputMinBlur,
        onKeyDown: handleMinKeyDown,
        className: `w-100 h-10 text-sm text-center border rounded-md ${isFilterApplied ? 'border-[hsl(var(--primary))] bg-primary/5 font-medium' : 'border  '}`
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement("span", {
        className: "mx-2 text-base font-bold text-filter-icon"
    }, "→"), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__["default"].createElement(__WEBPACK_EXTERNAL_MODULE__ui_input_js_dfab719b__.Input, {
        type: "text",
        value: inputMax,
        onChange: handleInputMaxChange,
        onBlur: handleInputMaxBlur,
        onKeyDown: handleMaxKeyDown,
        className: `w-100 h-10 text-sm text-center border rounded-md ${isFilterApplied ? 'border-[hsl(var(--primary))] bg-primary/5 font-medium' : 'border  '}`
    }))));
};
const SliderFilterRenderer_rslib_entry_ = SliderFilterRenderer;
export { SliderFilterRenderer_rslib_entry_ as default };
