// @cleartax/truss/dist/components/ui/accordion.js
import * as __WEBPACK_EXTERNAL_MODULE__radix_ui_react_accordion_496f1025__ from "@radix-ui/react-accordion";
import * as __WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__ from "lucide-react";
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__ from "../../lib/utils.js";
const Accordion = __WEBPACK_EXTERNAL_MODULE__radix_ui_react_accordion_496f1025__.Root;
const AccordionItem = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__radix_ui_react_accordion_496f1025__.Item, {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('border-b', className),
        ...props
    }));
AccordionItem.displayName = 'AccordionItem';
const AccordionTrigger = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, children, ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__radix_ui_react_accordion_496f1025__.Header, {
        className: "flex"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__radix_ui_react_accordion_496f1025__.Trigger, {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('flex flex-1 items-center justify-between py-4 text-sm font-medium transition-all hover:underline text-left [&[data-state=open]>svg]:rotate-180', className),
        ...props
    }, children, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.ChevronDown, {
        className: "h-4 w-4 shrink-0 text-muted-foreground transition-transform duration-200"
    }))));
AccordionTrigger.displayName = __WEBPACK_EXTERNAL_MODULE__radix_ui_react_accordion_496f1025__.Trigger.displayName;
const AccordionContent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, children, ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__radix_ui_react_accordion_496f1025__.Content, {
        ref: ref,
        className: "overflow-hidden text-sm data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down",
        ...props
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('pb-4 pt-0', className)
    }, children)));
AccordionContent.displayName = __WEBPACK_EXTERNAL_MODULE__radix_ui_react_accordion_496f1025__.Content.displayName;
export { Accordion, AccordionContent, AccordionItem, AccordionTrigger };
