// @cleartax/truss/dist/components/ui/calendar-v2.js
import * as __WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__ from "lucide-react";
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE_react_day_picker_7bc423fd__ from "react-day-picker";
import * as __WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__ from "./button.js";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__ from "../../lib/utils.js";
import * as __WEBPACK_EXTERNAL_MODULE__hooks_useCalendarV2_js_d1261b8a__ from "../../hooks/useCalendarV2.js";
import * as __WEBPACK_EXTERNAL_MODULE__monthSelector_js_631ecd06__ from "./monthSelector.js";
import * as __WEBPACK_EXTERNAL_MODULE__yearSelector_js_86300521__ from "./yearSelector.js";
function CalendarV2({ className, classNames, showOutsideDays = true, enableYearSelector = true, style = {}, ...props }) {
    const { view, selectedMonth, setSelectedMonth, yearRange, handleCaptionClick, handleYearClick, handleMonthSelect, handleYearSelect, handlePreviousYearRange, handleNextYearRange, handlePreviousMonthClick, handleNextMonthClick } = (0, __WEBPACK_EXTERNAL_MODULE__hooks_useCalendarV2_js_d1261b8a__.useCalendarV2)({
        defaultMonth: props.defaultMonth,
        enableYearSelector
    });
    const styleObject = {
        months: 'flex flex-col sm:flex-row space-y-4 sm:space-x-4 sm:space-y-0',
        month: 'space-y-4',
        caption: 'flex justify-center pt-1 pb-2 relative items-center',
        caption_label: 'text-sm font-medium ml-1',
        nav: 'space-x-1 flex items-center',
        nav_button: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)((0, __WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__.buttonVariants)({
            variant: 'outline'
        }), 'h-7 w-7 bg-transparent p-0 opacity-50 hover:opacity-100'),
        nav_button_previous: '',
        nav_button_next: '',
        table: 'w-full border-collapse space-y-1',
        head_row: 'flex pt-1',
        head_cell: 'text-muted-foreground rounded-md w-8 font-normal text-[0.8rem]',
        row: 'flex w-full mt-2',
        cell: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('relative p-0 text-center text-sm focus-within:relative focus-within:z-20 [&:has([aria-selected])]:bg-accent [&:has([aria-selected].day-outside)]:bg-accent/50 [&:has([aria-selected].day-range-end)]:rounded-r-md', 'range' === props.mode ? '[&:has(>.day-range-end)]:rounded-r-md [&:has(>.day-range-start)]:rounded-l-md first:[&:has([aria-selected])]:rounded-l-md last:[&:has([aria-selected])]:rounded-r-md' : '[&:has([aria-selected])]:rounded-md'),
        day: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)((0, __WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__.buttonVariants)({
            variant: 'ghost'
        }), 'h-8 w-8 p-0 font-normal aria-selected:opacity-100'),
        day_range_start: 'day-range-start',
        day_range_end: 'day-range-end',
        day_selected: 'bg-primary text-primary-foreground hover:bg-primary hover:text-primary-foreground focus:bg-primary focus:text-primary-foreground',
        day_today: 'bg-accent text-accent-foreground',
        day_outside: 'day-outside text-muted-foreground aria-selected:bg-accent/50 aria-selected:text-muted-foreground',
        day_disabled: 'text-muted-foreground opacity-50',
        day_range_middle: 'aria-selected:bg-accent aria-selected:text-accent-foreground',
        day_hidden: 'invisible',
        ...classNames
    };
    const CustomCaption = __WEBPACK_EXTERNAL_MODULE_react__.useCallback(({ displayMonth })=>{
        const currentDate = new Date();
        const currentYear = currentDate.getFullYear();
        const currentMonth = currentDate.getMonth();
        const month = displayMonth.toLocaleString('default', {
            month: 'long'
        });
        const year = displayMonth.getFullYear();
        return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
            className: "flex justify-center pt-1 pb-0 relative items-center"
        }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__.Button, {
            variant: "outline",
            className: "h-7 w-7 bg-transparent p-0 opacity-50 hover:opacity-100 mr-2",
            onClick: (e)=>{
                e.stopPropagation();
                handlePreviousMonthClick(displayMonth);
            },
            "aria-label": "Previous month"
        }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.ChevronLeft, {
            className: "h-4 w-4"
        })), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
            className: "text-sm font-medium cursor-pointer text-center mx-auto hover:text-gray-800 hover:bg-gray-100 px-2 py-1 rounded transition-colors",
            onClick: handleCaptionClick
        }, month, " ", year), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__.Button, {
            variant: "outline",
            className: "h-7 w-7 bg-transparent p-0 opacity-50 hover:opacity-100 ml-2",
            onClick: (e)=>{
                e.stopPropagation();
                handleNextMonthClick(displayMonth);
            },
            disabled: displayMonth.getFullYear() > currentYear || displayMonth.getFullYear() === currentYear && displayMonth.getMonth() >= currentMonth,
            "aria-label": "Next month"
        }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.ChevronRight, {
            className: "h-4 w-4"
        })));
    }, [
        handleCaptionClick,
        handlePreviousMonthClick,
        handleNextMonthClick
    ]);
    if ('month' === view) return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__monthSelector_js_631ecd06__.MonthSelector, {
        selectedMonth: selectedMonth,
        onMonthSelect: handleMonthSelect,
        onYearClick: handleYearClick,
        style: style,
        onSelectedMonthChange: setSelectedMonth
    });
    if ('year' === view) return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__yearSelector_js_86300521__.YearSelector, {
        selectedMonth: selectedMonth,
        yearRange: yearRange,
        onYearSelect: handleYearSelect,
        onPreviousYearRange: handlePreviousYearRange,
        onNextYearRange: handleNextYearRange,
        style: style
    });
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_react_day_picker_7bc423fd__.DayPicker, {
        showOutsideDays: showOutsideDays,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('p-3', className),
        classNames: styleObject,
        components: {
            IconLeft: ({ ...props })=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.ChevronLeft, {
                    className: "h-4 w-4"
                }),
            IconRight: ({ ...props })=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.ChevronRight, {
                    className: "h-4 w-4"
                }),
            Caption: enableYearSelector ? CustomCaption : void 0
        },
        month: selectedMonth,
        onMonthChange: setSelectedMonth,
        style: style,
        ...props
    });
}
CalendarV2.displayName = 'CalendarV2';
export { CalendarV2 };
