// @cleartax/truss/dist/components/ui/carousel.js
import * as __WEBPACK_EXTERNAL_MODULE_embla_carousel_react_7f67003b__ from "embla-carousel-react";
import * as __WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__ from "lucide-react";
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__ from "./button.js";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__ from "../../lib/utils.js";
const CarouselContext = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createContext(null);
function useCarousel() {
    const context = __WEBPACK_EXTERNAL_MODULE_react__.useContext(CarouselContext);
    if (!context) throw new Error('useCarousel must be used within a <Carousel />');
    return context;
}
const Carousel = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ orientation = 'horizontal', opts, setApi, plugins, className, children, ...props }, ref)=>{
    const [carouselRef, api] = (0, __WEBPACK_EXTERNAL_MODULE_embla_carousel_react_7f67003b__["default"])({
        ...opts,
        axis: 'horizontal' === orientation ? 'x' : 'y'
    }, plugins);
    const [canScrollPrev, setCanScrollPrev] = __WEBPACK_EXTERNAL_MODULE_react__.useState(false);
    const [canScrollNext, setCanScrollNext] = __WEBPACK_EXTERNAL_MODULE_react__.useState(false);
    const onSelect = __WEBPACK_EXTERNAL_MODULE_react__.useCallback((api)=>{
        if (!api) return;
        setCanScrollPrev(api.canScrollPrev());
        setCanScrollNext(api.canScrollNext());
    }, []);
    const scrollPrev = __WEBPACK_EXTERNAL_MODULE_react__.useCallback(()=>{
        api?.scrollPrev();
    }, [
        api
    ]);
    const scrollNext = __WEBPACK_EXTERNAL_MODULE_react__.useCallback(()=>{
        api?.scrollNext();
    }, [
        api
    ]);
    const handleKeyDown = __WEBPACK_EXTERNAL_MODULE_react__.useCallback((event)=>{
        if ('ArrowLeft' === event.key) {
            event.preventDefault();
            scrollPrev();
        } else if ('ArrowRight' === event.key) {
            event.preventDefault();
            scrollNext();
        }
    }, [
        scrollPrev,
        scrollNext
    ]);
    __WEBPACK_EXTERNAL_MODULE_react__.useEffect(()=>{
        if (!api || !setApi) return;
        setApi(api);
    }, [
        api,
        setApi
    ]);
    __WEBPACK_EXTERNAL_MODULE_react__.useEffect(()=>{
        if (!api) return;
        onSelect(api);
        api.on('reInit', onSelect);
        api.on('select', onSelect);
        return ()=>{
            api?.off('select', onSelect);
        };
    }, [
        api,
        onSelect
    ]);
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(CarouselContext.Provider, {
        value: {
            carouselRef,
            api: api,
            opts,
            orientation: orientation || (opts?.axis === 'y' ? 'vertical' : 'horizontal'),
            scrollPrev,
            scrollNext,
            canScrollPrev,
            canScrollNext
        }
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        ref: ref,
        onKeyDownCapture: handleKeyDown,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('relative', className),
        role: "region",
        "aria-roledescription": "carousel",
        ...props
    }, children));
});
Carousel.displayName = 'Carousel';
const CarouselContent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, ...props }, ref)=>{
    const { carouselRef, orientation } = useCarousel();
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        ref: carouselRef,
        className: "overflow-hidden"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('flex', 'horizontal' === orientation ? '-ml-4' : '-mt-4 flex-col', className),
        ...props
    }));
});
CarouselContent.displayName = 'CarouselContent';
const CarouselItem = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, ...props }, ref)=>{
    const { orientation } = useCarousel();
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        ref: ref,
        role: "group",
        "aria-roledescription": "slide",
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('min-w-0 shrink-0 grow-0 basis-full', 'horizontal' === orientation ? 'pl-4' : 'pt-4', className),
        ...props
    });
});
CarouselItem.displayName = 'CarouselItem';
const CarouselPrevious = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, variant = 'outline', size = 'icon', ...props }, ref)=>{
    const { orientation, scrollPrev, canScrollPrev } = useCarousel();
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__.Button, {
        ref: ref,
        variant: variant,
        size: size,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('absolute  h-8 w-8 rounded-full', 'horizontal' === orientation ? '-left-12 top-1/2 -translate-y-1/2' : '-top-12 left-1/2 -translate-x-1/2 rotate-90', className),
        disabled: !canScrollPrev,
        onClick: scrollPrev,
        ...props
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.ArrowLeft, {
        className: "h-4 w-4"
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", {
        className: "sr-only"
    }, "Previous slide"));
});
CarouselPrevious.displayName = 'CarouselPrevious';
const CarouselNext = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, variant = 'outline', size = 'icon', ...props }, ref)=>{
    const { orientation, scrollNext, canScrollNext } = useCarousel();
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__.Button, {
        ref: ref,
        variant: variant,
        size: size,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('absolute h-8 w-8 rounded-full', 'horizontal' === orientation ? '-right-12 top-1/2 -translate-y-1/2' : '-bottom-12 left-1/2 -translate-x-1/2 rotate-90', className),
        disabled: !canScrollNext,
        onClick: scrollNext,
        ...props
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.ArrowRight, {
        className: "h-4 w-4"
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", {
        className: "sr-only"
    }, "Next slide"));
});
CarouselNext.displayName = 'CarouselNext';
export { Carousel, CarouselContent, CarouselItem, CarouselNext, CarouselPrevious };
