// @cleartax/truss/dist/components/ui/drawer.js
'use client';
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE_vaul__ from "vaul";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__ from "../../lib/utils.js";
const Drawer = ({ shouldScaleBackground = true, direction = 'bottom', ...props })=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_vaul__.Drawer.Root, {
        shouldScaleBackground: shouldScaleBackground,
        direction: direction,
        ...props
    });
Drawer.displayName = 'Drawer';
const DrawerTrigger = __WEBPACK_EXTERNAL_MODULE_vaul__.Drawer.Trigger;
const DrawerPortal = __WEBPACK_EXTERNAL_MODULE_vaul__.Drawer.Portal;
const DrawerClose = __WEBPACK_EXTERNAL_MODULE_vaul__.Drawer.Close;
const DrawerOverlay = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_vaul__.Drawer.Overlay, {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('fixed inset-0 z-50 bg-black/80', className),
        ...props
    }));
DrawerOverlay.displayName = __WEBPACK_EXTERNAL_MODULE_vaul__.Drawer.Overlay.displayName;
const DrawerContent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, children, direction = 'bottom', ...props }, ref)=>{
    const directionClasses = {
        top: 'fixed inset-x-0 top-0 z-50 mb-24 flex h-auto flex-col rounded-b-[10px] border bg-background',
        right: 'fixed inset-y-0 right-0 z-50 ml-24 flex h-full w-auto flex-row rounded-l-[10px] border bg-background',
        bottom: 'fixed inset-x-0 bottom-0 z-50 mt-24 flex h-auto flex-col rounded-t-[10px] border bg-background',
        left: 'fixed inset-y-0 left-0 z-50 mr-24 flex h-full w-auto flex-row rounded-r-[10px] border bg-background'
    };
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(DrawerPortal, null, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(DrawerOverlay, null), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_vaul__.Drawer.Content, {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)(directionClasses[direction], className),
        ...props
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('bottom' === direction ? "mx-auto mt-4 h-2 w-[100px] rounded-full bg-muted" : 'top' === direction ? "mx-auto mb-4 h-2 w-[100px] rounded-full bg-muted" : 'left' === direction ? "my-auto mr-4 h-[100px] w-2 rounded-full bg-muted" : "my-auto ml-4 h-[100px] w-2 rounded-full bg-muted")
    }), children));
});
DrawerContent.displayName = 'DrawerContent';
const DrawerHeader = ({ className, ...props })=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('grid gap-1.5 p-4 text-center sm:text-left', className),
        ...props
    });
DrawerHeader.displayName = 'DrawerHeader';
const DrawerFooter = ({ className, ...props })=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('mt-auto flex flex-col gap-2 p-4', className),
        ...props
    });
DrawerFooter.displayName = 'DrawerFooter';
const DrawerTitle = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_vaul__.Drawer.Title, {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('text-lg font-semibold leading-none tracking-tight', className),
        ...props
    }));
DrawerTitle.displayName = __WEBPACK_EXTERNAL_MODULE_vaul__.Drawer.Title.displayName;
const DrawerDescription = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_vaul__.Drawer.Description, {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('text-sm text-muted-foreground', className),
        ...props
    }));
DrawerDescription.displayName = __WEBPACK_EXTERNAL_MODULE_vaul__.Drawer.Description.displayName;
export { Drawer, DrawerClose, DrawerContent, DrawerDescription, DrawerFooter, DrawerHeader, DrawerOverlay, DrawerPortal, DrawerTitle, DrawerTrigger };
