// @cleartax/truss/dist/components/ui/input-otp.js
import * as __WEBPACK_EXTERNAL_MODULE_input_otp_9f71df26__ from "input-otp";
import * as __WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__ from "lucide-react";
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__ from "../../lib/utils.js";
const InputOTP = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, containerClassName, ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_input_otp_9f71df26__.OTPInput, {
        ref: ref,
        containerClassName: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('flex items-center gap-2 has-[:disabled]:opacity-50', containerClassName),
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('disabled:cursor-not-allowed', className),
        ...props
    }));
InputOTP.displayName = 'InputOTP';
const InputOTPGroup = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('flex items-center', className),
        ...props
    }));
InputOTPGroup.displayName = 'InputOTPGroup';
const InputOTPSlot = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ index, className, ...props }, ref)=>{
    const inputOTPContext = __WEBPACK_EXTERNAL_MODULE_react__.useContext(__WEBPACK_EXTERNAL_MODULE_input_otp_9f71df26__.OTPInputContext);
    const { char, hasFakeCaret, isActive } = inputOTPContext.slots[index];
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('relative flex h-9 w-9 items-center justify-center border-y border-r border-input text-sm shadow-sm transition-all first:rounded-l-md first:border-l last:rounded-r-md', isActive && 'z-10 ring-1 ring-ring', className),
        ...props
    }, char, hasFakeCaret && /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: "pointer-events-none absolute inset-0 flex items-center justify-center"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: "h-4 w-px animate-caret-blink bg-foreground duration-1000"
    })));
});
InputOTPSlot.displayName = 'InputOTPSlot';
const InputOTPSeparator = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        ref: ref,
        role: "separator",
        ...props
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.Minus, null)));
InputOTPSeparator.displayName = 'InputOTPSeparator';
export { InputOTP, InputOTPGroup, InputOTPSeparator, InputOTPSlot };
