// @cleartax/truss/dist/components/ui/pagination.js
import * as __WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__ from "lucide-react";
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__ from "./button.js";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__ from "../../lib/utils.js";
const Pagination = ({ className, ...props })=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("nav", {
        role: "navigation",
        "aria-label": "pagination",
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('mx-auto flex w-full justify-center', className),
        ...props
    });
Pagination.displayName = 'Pagination';
const PaginationContent = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("ul", {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('flex flex-row items-center gap-1', className),
        ...props
    }));
PaginationContent.displayName = 'PaginationContent';
const PaginationItem = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("li", {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('', className),
        ...props
    }));
PaginationItem.displayName = 'PaginationItem';
const PaginationLink = ({ className, isActive, size = 'icon', ...props })=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("a", {
        "aria-current": isActive ? 'page' : void 0,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)((0, __WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__.buttonVariants)({
            variant: isActive ? 'outline' : 'ghost',
            size
        }), className),
        ...props
    });
PaginationLink.displayName = 'PaginationLink';
const PaginationPrevious = ({ className, ...props })=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(PaginationLink, {
        "aria-label": "Go to previous page",
        size: "default",
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('gap-1 pl-2.5', className),
        ...props
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.ChevronLeft, {
        className: "h-4 w-4"
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", null, "Previous"));
PaginationPrevious.displayName = 'PaginationPrevious';
const PaginationNext = ({ className, ...props })=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(PaginationLink, {
        "aria-label": "Go to next page",
        size: "default",
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('gap-1 pr-2.5', className),
        ...props
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", null, "Next"), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.ChevronRight, {
        className: "h-4 w-4"
    }));
PaginationNext.displayName = 'PaginationNext';
const PaginationEllipsis = ({ className, ...props })=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", {
        "aria-hidden": true,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('flex h-9 w-9 items-center justify-center', className),
        ...props
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.MoreHorizontal, {
        className: "h-4 w-4"
    }), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", {
        className: "sr-only"
    }, "More pages"));
PaginationEllipsis.displayName = 'PaginationEllipsis';
export { Pagination, PaginationContent, PaginationEllipsis, PaginationItem, PaginationLink, PaginationNext, PaginationPrevious };
