// @cleartax/truss/dist/components/ui/sonner.js
'use client';
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE_next_themes_25055421__ from "next-themes";
import * as __WEBPACK_EXTERNAL_MODULE_sonner__ from "sonner";
const Toaster = ({ ...props })=>{
    const { theme = 'system' } = (0, __WEBPACK_EXTERNAL_MODULE_next_themes_25055421__.useTheme)();
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_sonner__.Toaster, {
        theme: theme,
        className: "toaster group",
        toastOptions: {
            classNames: {
                toast: 'group toast group-[.toaster]:bg-background group-[.toaster]:text-foreground group-[.toaster]:border group-[.toaster]:shadow-lg',
                description: 'group-[.toast]:text-muted-foreground',
                actionButton: 'group-[.toast]:bg-primary group-[.toast]:text-primary-foreground',
                cancelButton: 'group-[.toast]:bg-muted group-[.toast]:text-muted-foreground'
            }
        },
        ...props
    });
};
export { Toaster };
