// @cleartax/truss/dist/components/ui/toggle-group.js
import * as __WEBPACK_EXTERNAL_MODULE__radix_ui_react_toggle_group_4f42c91c__ from "@radix-ui/react-toggle-group";
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__toggle_js_10734c63__ from "./toggle.js";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__ from "../../lib/utils.js";
const ToggleGroupContext = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createContext({
    size: 'default',
    variant: 'default'
});
const ToggleGroup = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, variant, size, children, ...props }, ref)=>/*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__radix_ui_react_toggle_group_4f42c91c__.Root, {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('flex items-center justify-center gap-1', className),
        ...props
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(ToggleGroupContext.Provider, {
        value: {
            variant,
            size
        }
    }, children)));
ToggleGroup.displayName = __WEBPACK_EXTERNAL_MODULE__radix_ui_react_toggle_group_4f42c91c__.Root.displayName;
const ToggleGroupItem = /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.forwardRef(({ className, children, variant, size, ...props }, ref)=>{
    const context = __WEBPACK_EXTERNAL_MODULE_react__.useContext(ToggleGroupContext);
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__radix_ui_react_toggle_group_4f42c91c__.Item, {
        ref: ref,
        className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)((0, __WEBPACK_EXTERNAL_MODULE__toggle_js_10734c63__.toggleVariants)({
            variant: context.variant || variant,
            size: context.size || size
        }), className),
        ...props
    }, children);
});
ToggleGroupItem.displayName = __WEBPACK_EXTERNAL_MODULE__radix_ui_react_toggle_group_4f42c91c__.Item.displayName;
export { ToggleGroup, ToggleGroupItem };
