// @cleartax/truss/dist/components/ui/yearSelector.js
import * as __WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__ from "lucide-react";
import * as __WEBPACK_EXTERNAL_MODULE_react__ from "react";
import * as __WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__ from "./button.js";
import * as __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__ from "../../lib/utils.js";
function YearSelector({ selectedMonth, yearRange, onYearSelect, onPreviousYearRange, onNextYearRange, style = {} }) {
    const [startYear, endYear] = yearRange;
    const years = Array.from({
        length: 12
    }, (_, i)=>startYear + i);
    const selectedYear = selectedMonth.getFullYear();
    const currentDate = new Date();
    const currentYear = currentDate.getFullYear();
    return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: "p-2",
        style: style
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: "flex justify-center relative items-center"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__.Button, {
        variant: "outline",
        className: "h-7 w-7 bg-transparent p-0 opacity-50 hover:opacity-100 mr-2",
        onClick: onPreviousYearRange,
        "aria-label": "Previous year range"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.ChevronLeft, {
        className: "h-4 w-4"
    })), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: "text-sm font-medium text-center mx-auto hover:text-gray-800 hover:bg-gray-100 px-2 py-1 rounded transition-colors cursor-pointer"
    }, startYear, "-", endYear), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__.Button, {
        variant: "outline",
        className: "h-7 w-7 bg-transparent p-0 opacity-50 hover:opacity-100 ml-2",
        onClick: onNextYearRange,
        disabled: endYear >= currentYear,
        "aria-label": "Next year range"
    }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE_lucide_react_f128bbbb__.ChevronRight, {
        className: "h-4 w-4"
    }))), /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("div", {
        className: "mt-3 grid grid-cols-3"
    }, years.map((year)=>{
        const isFutureYear = year > currentYear;
        return /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement(__WEBPACK_EXTERNAL_MODULE__button_js_821b8d9d__.Button, {
            key: year,
            variant: "ghost",
            className: (0, __WEBPACK_EXTERNAL_MODULE__lib_utils_js_c09d30d7__.cn)('h-8 w-full justify-center transition-colors text-xs', selectedYear === year && 'bg-primary/80 text-primary-foreground hover:bg-primary/80 hover:text-primary-foreground', selectedYear !== year && !isFutureYear && 'hover:bg-accent hover:text-accent-foreground hover:scale-105', isFutureYear && 'opacity-40 cursor-not-allowed'),
            onClick: ()=>!isFutureYear && onYearSelect(year),
            disabled: isFutureYear
        }, /*#__PURE__*/ __WEBPACK_EXTERNAL_MODULE_react__.createElement("span", {
            className: "px-0.5"
        }, year));
    })));
}
YearSelector.displayName = 'YearSelector';
export { YearSelector };
